presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -438,13 +439,40 @@ protected TupleDescriptor visitInsert(Insert insert, AnalysisContext context)
         accessControl.checkCanInsertIntoTable(session.getIdentity(), targetTable);
         analysis.setInsertTarget(targetTableHandle.get());
 
-        List<ColumnMetadata> columns = metadata.getTableMetadata(session, targetTableHandle.get()).getColumns();
-        Iterable<Type> tableTypes = columns.stream()
-                .filter(column -> !column.isHidden())
-                .map(ColumnMetadata::getType)
-                .collect(toImmutableList());
+        List<ColumnMetadata> allTableColumns = metadata.getTableMetadata(session, targetTableHandle.get()).getColumns();
+        final Iterable<Type> tableTypes;
+        if (insert.getColumns().isPresent()) {
```

Comment:
maybe, instead having `if/else` block, always execute code for insertColumns present.
But when those are not present treat them as all visible columns from target table.
```
List<String> insertColumns = analysis.getInsertColumns().orElse(allTableColumnsNames)
```